### PR TITLE
Contract Spec XDR types - Part 2

### PR DESCRIPTION
**Pull request by CristhianRodriguezMolina**
Contributing to #228 

In this PR, the following types were added:

- SCSpecUDTEnumCaseV0List
- SCSpecUDTEnumV0
- SCSpecUDTErrorEnumCaseV0List
- SCSpecUDTErrorEnumV0

### CODE REVIEW

File: lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex [hunk 1]
```diff
@@ -0,0 +1,121 @@
+defmodule StellarBase.XDR.SCSpecUDTEnumV0 do
+  @moduledoc """
+  Representation of Stellar's ledger SCSpecUDTEnumV0
+  """
```
Comment (miguelnietoa) [inline lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex:4]: Is it right?

File: lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        %String80{} = lib,
+        %String60{} = name,
+        %SCSpecUDTEnumCaseV0List{} = cases
+      ),
+      do: %__MODULE__{
+        doc: doc,
+        lib: lib,
+        name: name,
+        cases: cases
+      }
```
Comment (miguelnietoa) [inline lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex:43]: Let's keep this in one line!

File: lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        lib: lib,
+        name: name,
+        cases: cases
+      }) do
+    [
+      doc: doc,
+      lib: lib,
+      name: name,
+      cases: cases
+    ]
```
Comment (miguelnietoa) [inline lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex:57]: Same here

File: lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
```
Comment (miguelnietoa) [inline lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex:74]: Same here

File: lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            cases: cases
+          ]
+        }, rest}} ->
+        {:ok,
+         {new(
+            doc,
+            lib,
+            name,
+            cases
+          ), rest}}
```
Comment (miguelnietoa) [inline lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex:99]: I think you guys could write this code less split!

**Review by miguelnietoa — CHANGES_REQUESTED** [review]

File: lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex (hunk 1 above)
Comment (CristhianRodriguezMolina) [reply lib/xdr/contract/spec/sc_spec_udt_enum_v0.ex:4]: No, it's not a ledger, we'll fix it.

**Review by miguelnietoa — APPROVED** [review]
Awesome, go ahead! 🚀

**Pull request merged**